test/metadata_test.go

```diff
@@ -93,29 +122,33 @@ func (s) TestInvalidMetadata(t *testing.T) {
 	defer ss.Stop()
 
 	for _, test := range tests {
-		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-		defer cancel()
-
-		ctx = metadata.NewOutgoingContext(ctx, test.md)
-		if _, err := ss.Client.EmptyCall(ctx, &testpb.Empty{}); !reflect.DeepEqual(test.want, err) {
-			t.Errorf("call ss.Client.EmptyCall() validate metadata which is %v got err :%v, want err :%v", test.md, err, test.want)
-		}
+		t.Run("unary "+test.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+			defer cancel()
+			ctx = metadata.NewOutgoingContext(ctx, test.md)
+			ctx = metadata.AppendToOutgoingContext(ctx, test.appendMD...)
+			if _, err := ss.Client.EmptyCall(ctx, &testpb.Empty{}); !reflect.DeepEqual(test.want, err) {
+				t.Errorf("call ss.Client.EmptyCall() validate metadata which is %v got err :%v, want err :%v", test.md, err, test.want)
+			}
+		})
 	}
 
 	// call the stream server's api to drive the server-side unit testing
 	for _, test := range tests {
-		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-		stream, err := ss.Client.FullDuplexCall(ctx)
-		defer cancel()
-		if err != nil {
-			t.Errorf("call ss.Client.FullDuplexCall(context.Background()) will success but got err :%v", err)
-			continue
-		}
-		if err := stream.Send(&testpb.StreamingOutputCallRequest{}); err != nil {
-			t.Errorf("call ss.Client stream Send(nil) will success but got err :%v", err)
-		}
-		if _, err := stream.Recv(); status.Code(err) != status.Code(test.recv) || !strings.Contains(err.Error(), test.recv.Error()) {
-			t.Errorf("stream.Recv() = _, get err :%v, want err :%v", err, test.recv)
-		}
+		t.Run("streaming "+test.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+			defer cancel()
+			stream, err := ss.Client.FullDuplexCall(ctx)
+			if err != nil {
+				t.Errorf("call ss.Client.FullDuplexCall(context.Background()) will success but got err :%v", err)
+				return
+			}
+			if err := stream.Send(&testpb.StreamingOutputCallRequest{}); err != nil {
+				t.Errorf("call ss.Client stream Send(nil) will success but got err :%v", err)
```

Comment:
Same here (see comment above). This isn't sending nil, but an allocated testpb.StreamingOutputCallRequest.